pkg/manager/manager.go

```
@@ -781,8 +1041,490 @@ func SyncLocalRules(cursorDir string) error {
+
+// ShareableLock represents the structure of the shareable lockfile.
+type ShareableLock struct {
+	// Version of the shareable file format
+	FormatVersion int `json:"formatVersion"`
+
+	// Rules that can be shared
+	Rules []ShareableRule `json:"rules"`
+}
+
+// The shareable file excludes personal data like local paths.
+func ShareRules(cursorDir string, shareFilePath string, embedContent bool) error {
```

golangci-lint (1.64.8)

cognitive complexity 33 of func `ShareRules` is high (> 20)

(gocognit)